### PR TITLE
Clean up main configure.ac and autotools

### PR DESCRIPTION
**Pull request by grimreaper**
- update config.guess and config.sub
- remove obsolescent macros
- generally assume we're running on a platform produced in the last 20 years

### CODE REVIEW

File: src/config.h.in [hunk 1]
```diff
@@ -106,21 +99,12 @@
 /* Define to 1 if you have the <limits.h> header file. */
 #undef HAVE_LIMITS_H
 
-/* Define to 1 if you have the `lockf' function. */
-#undef HAVE_LOCKF
```
Comment (jmroot) [inline src/config.h.in]: HAVE_LOCKF is checked in [adv-flock.c](https://github.com/macports/macports-base/blob/master/src/pextlib1.0/adv-flock.c#L68).

File: src/config.h.in [hunk 2]
```diff
@@ -187,9 +171,6 @@
 /* Define to 1 if you have the <stdlib.h> header file. */
 #undef HAVE_STDLIB_H
 
-/* Define to 1 if you have the `strcasecmp' function. */
-#undef HAVE_STRCASECMP
```
Comment (jmroot) [inline src/config.h.in]: HAVE_STRCASECMP and HAVE_STRNCASECMP are checked in [strcasecmp.h](https://github.com/macports/macports-base/blob/master/src/pextlib1.0/strcasecmp.h#L39).

File: configure.ac [hunk 3]
```diff
@@ -109,7 +109,6 @@ dnl example, these include BZIP2 and GZIP, which is the reason why they are
 dnl named with a _BIN suffix here.
 MP_TOOL_PATH(BSDMAKE, [bsdmake/pmake])
 MP_TOOL_PATH(BZIP2_BIN, [bzip2])
-MP_TOOL_PATH(CVS, [cvs])
```
[the remark this answers is not in the document]
Comment (grimreaper) [reply configure.ac]: Where? I grepped around but could not find any reference to them.

File: src/config.h.in (hunk 1 above)
Comment (grimreaper) [reply src/config.h.in]: this was an auto-gen change - forgot to check for defines :\ - thoughts on just removing the HAVE check?

File: src/config.h.in (hunk 2 above)
Comment (grimreaper) [reply src/config.h.in]: I think the correct fix is to just remove that file (along with the implementation). am I wrong?1

File: configure.ac (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmroot) [reply configure.ac]: https://github.com/macports/macports-base/blob/master/src/port1.0/port_autoconf.tcl.in

File: src/config.h.in (hunk 1 above)
Comment (jmroot) [reply src/config.h.in]: Probably fine if lockf is going to exist on any reasonable target platform. It's used as a fallback if flock isn't available.

File: src/config.h.in (hunk 2 above)
Comment (jmroot) [reply src/config.h.in]: Looks like it.

**Comment by grimreaper** [comment]
updated. 
not for this diff, but i wonder if we should just always use lockf considering its in POSIX.

File: src/pextlib1.0/filemap.c [hunk 4]
```diff
@@ -45,7 +45,6 @@
 #include <tcl.h>
 
 #include "filemap.h"
-#include "strcasecmp.h"
```
Comment (jmroot) [inline src/pextlib1.0/filemap.c]: This file now needs to `#include <strings.h>`.

Comment (grimreaper) [reply src/pextlib1.0/filemap.c]: bah. you're right. compile tested but the test system has terrible header pollution

Comment (grimreaper) [reply src/pextlib1.0/filemap.c]: also adding string.h for memcmp

**Pull request merged**